lib/src/cpp/domain_participant/DefaultExternalUnicastLocators.cpp

```diff
@@ -235,8 +235,7 @@ void set_port(
         rtps_node->appendChild(locator_list_node);
     }
 
-    common::locator_list::set_port(*manager, *dynamic_cast<xercesc::DOMElement*>(locator_list_node), port, index,
-            true);
+    common::locator_list::set_port(*manager, locator_list_node, port, index, true);
```

Comment:
I think that in the infrastructure refactor, the manager does not require to be a pointer.